src/tallies/tally_filter_header.F90

```diff
@@ -104,4 +129,95 @@ subroutine filter_initialize(this)
     class(TallyFilter), intent(inout) :: this
   end subroutine filter_initialize
 
+!===============================================================================
+!                               C API FUNCTIONS
+!===============================================================================
+
+  function openmc_extend_filters(n, index_start, index_end) result(err) bind(C)
+    ! Creates or extends the filters array
+    integer(C_INT32_T), value, intent(in) :: n
+    integer(C_INT32_T), optional, intent(out) :: index_start
+    integer(C_INT32_T), optional, intent(out) :: index_end
+    integer(C_INT) :: err
+
+    integer :: i ! loop counter
+    type(TallyFilterContainer), allocatable :: temp(:) ! temporary filters
+
+    if (n_filters == 0) then
+      ! Allocate filters array
+      allocate(filters(n))
+    else
+      ! Move filters to temporary array
+      allocate(temp(n_filters + n))
+      do i = 1, n_filters
+        call move_alloc(filters(i) % obj, temp(i) % obj)
+      end do
+
+      ! Move filters back from temporary array to filters array
+      call move_alloc(temp, filters)
+    end if
+
+    ! Return indices in filters array
+    if (present(index_start)) index_start = n_filters + 1
+    if (present(index_end)) index_end = n_filters + n
+    n_filters = n_filters + n
+
+    err = 0
+  end function openmc_extend_filters
+
+
+  function openmc_filter_get_id(index, id) result(err) bind(C)
+    ! Return the ID of a filter
+    integer(C_INT32_T), value       :: index
+    integer(C_INT32_T), intent(out) :: id
+    integer(C_INT) :: err
+
+    if (index >= 1 .and. index <= n_filters) then
+      id = filters(index) % obj % id
+      err = 0
+    else
+      err = E_OUT_OF_BOUNDS
+    end if
+  end function openmc_filter_get_id
+
+
+  function openmc_filter_set_id(index, id) result(err) bind(C)
+    ! Set the ID of a filter
+    integer(C_INT32_T), value, intent(in) :: index
+    integer(C_INT32_T), value, intent(in) :: id
+    integer(C_INT) :: err
+
+    if (index >= 1 .and. index <= n_filters) then
+      if (allocated(filters(index) % obj)) then
+        filters(index) % obj % id = id
+        call filter_dict % add_key(id, index)
+
+        err = 0
+      else
+        err = E_FILTER_NOT_ALLOCATED
+      end if
+    else
+      err = E_OUT_OF_BOUNDS
+    end if
+  end function openmc_filter_set_id
+
+
+  function openmc_get_filter(id, index) result(err) bind(C)
```

Comment:
`openmc_index_filter` would be a more specific name.

Similarly, `openmc_index_tally`, `openmc_index_cell`, ...

Reply:
I've renamed these to `openmc_get_*_index`.